plugin/acl/setup.go

```diff
@@ -0,0 +1,188 @@
+package acl
+
+import (
+	"net"
+	"strings"
+
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/metrics"
+
+	"github.com/caddyserver/caddy"
+	"github.com/infobloxopen/go-trees/iptree"
+	"github.com/miekg/dns"
+)
+
+const (
+	// QtypeAll is used to match any kinds of DNS records type.
+	// NOTE: The value of QtypeAll should be different with other QTYPEs defined in miekg/dns.
+	QtypeAll uint16 = 299
+)
+
+var (
+	// DefaultFilter is the default ip filter which matches both IPv4All and IPv6All.
+	DefaultFilter *iptree.Tree
```

Comment:
does this cache things? Because with udp and ipv6 support it is trivial to memory DoS the server by sending a bunch of bogus traffic.

Reply:
No, it won't cache anything at runtime. The filter we use will only update in setup.